### DISCUSSION TITLE
Missing feature - hyperlinks

### DISCUSSION THREAD

**Discussion by marvinsmarvin** [pwndoc/pwndoc#386] [Ideas]
I am looking for some to collaborate with in adding hyperlinks into this tool. If anyone can help I'll be glad.
I found this module: https://www.npmjs.com/package/gasperoid-docxtemplater-link-module. Maybe this can be helpful.

**Comment by RedTeamer69** [comment]
Hi,

I am missing this feature too.  
I have tried to install docxtemplater-link-module to backend. It is possible and it should work.
Fronted uses Tiptap editor, Tiptap has support for links - link extension https://tiptap.dev/api/marks/link.
I guess we need: 
- enable links in Tiptap editor
- install docxtemplater-link-module
- add code to backend which can handle links 

Am I correct? Help is appreciated. I am just pen tester not a developer.
Thanks

**Comment by Zeecka** [comment]
Hi, [WhiteWinterWolf](https://github.com/WhiteWinterWolf) implemented links on customfilters in PR https://github.com/pwndoc/pwndoc/pull/281 .

I also wanted to dig on TipTap WYSIWYG but the fact is that PwnDoc use an old major version of TipTap vailable [here](https://github.com/ueberdosis/tiptap/tree/v1). The "Link" frontend extension seems to be available on that version (see [Available Extensions](https://github.com/ueberdosis/tiptap/tree/v1#available-extensions)) but must be reimplemented with Docxtemplater, like in WhiteWinterWolf PR. 

The frontend plugin should be added in editor.vue. https://github.com/pwndoc/pwndoc/blob/265e359cf2383794ce411193e3526270cbae8902/frontend/src/components/editor.vue#L199-L214

**Reply by RedTeamer69** [reply]
Hi,
thanks for the inspiration, it was constructive. Now I have limited - partial - statical support for hypertext link.

I've copied, editor.vue setting from https://github.com/pwndoc/pwndoc/pull/208. It enables link support in frontend editor.
Then I've put custom filters linkTo and p from https://github.com/pwndoc/pwndoc/pull/281 to custom-generator.js. 
 
If I use tag linkTo in template, hypertext link is working, but there is statical field with link in template. 
Example -  very useful when you want to link external CVSS calculator
{@cvss.vectorString | linkTo: 'https://www.first.org/cvss/calculator/3.1#' + cvss.vectorString | p}

Unfortunately, it not works, without statical URL part in tag.

I am attaching modified files (editor.vue, custom-generator.js) . BUT I had to skip htmlEncode function in editor.vue (it converted HTML <a> to entities and I was not able to handle it correctly). Using it is a security risk.

Help and comments are appreciated
Thanks
[link-support.zip](https://github.com/pwndoc/pwndoc/files/10084575/link-support.zip)
